catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/resource/download/ReliableResourceDownloadManager.java

```diff
@@ -200,21 +210,100 @@ public void setProductCacheDirectory(String productCacheDirectory) {
         return downloadsInProgress;
     }
 
-    private ResourceResponse startDownload(String downloadIdentifier,
-            ResourceResponse resourceResponse, ResourceRetriever retriever, Metacard metacard) {
+    public void cleanUp() {
+        executor.shutdown();
+        try {
+            executor.awaitTermination(ONE_SECOND_IN_MS, TimeUnit.MILLISECONDS);
+        } catch (InterruptedException e) {
+            executor.shutdownNow();
+        }
+    }
+
+    private void publishStartEvent(Metacard metacard, ResourceResponse resourceResponse, String downloadIdentifier) {
+        downloaderConfig.getEventPublisher()
+                .postRetrievalStatus(resourceResponse,
+                        ProductRetrievalStatus.STARTED,
+                        metacard,
+                        null,
+                        0L,
+                        downloadIdentifier);
+    }
+
+    private ResourceResponse startDownload(Metacard metacard, ResourceRetriever retriever,
```

Comment:
This method has gotten pretty long compared to the original. May be worth looking at how we could split it up.